File: tests/testdata/codegen/deferred-outputs-pp/nodejs/first.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as random from "@codeinfra/random";

interface FirstArgs {
    passwordLength: codeinfra.Input<number>,
}

export class First extends codeinfra.ComponentResource {
    public petName: codeinfra.Output<string>;
    public password: codeinfra.Output<string>;
    constructor(name: string, args: FirstArgs, opts?: codeinfra.ComponentResourceOptions) {
        super("components:index:First", name, args, opts);
        const randomPet = new random.RandomPet(`${name}-randomPet`, {}, {
            parent: this,
        });

        const randomPassword = new random.RandomPassword(`${name}-randomPassword`, {length: args.passwordLength}, {
            parent: this,
        });

        this.petName = randomPet.id;
        this.password = randomPassword.result;
        this.registerOutputs({
            petName: randomPet.id,
            password: randomPassword.result,
        });
    }
}

File: tests/testdata/codegen/deferred-outputs-pp/python/first.py
import codeinfra
from codeinfra import Input
from typing import Optional, Dict, TypedDict, Any
import codeinfra_random as random

class FirstArgs(TypedDict, total=False):
    passwordLength: Input[float]

class First(codeinfra.ComponentResource):
    def __init__(self, name: str, args: FirstArgs, opts:Optional[codeinfra.ResourceOptions] = None):
        super().__init__("components:index:First", name, args, opts)

        random_pet = random.RandomPet(f"{name}-randomPet", opts = codeinfra.ResourceOptions(parent=self))

        random_password = random.RandomPassword(f"{name}-randomPassword", length=args["passwordLength"],
        opts = codeinfra.ResourceOptions(parent=self))

        self.petName = random_pet.id
        self.password = random_password.result
        self.register_outputs({
            'petName': random_pet.id, 
            'password': random_password.result
        })